src/views/virtualmachines/details/components/snapshots/hooks/useSnapshotData.ts

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+
+import {
+  VirtualMachineRestoreModelGroupVersionKind,
+  VirtualMachineSnapshotModelGroupVersionKind,
+} from '@kubevirt-ui/kubevirt-api/console';
+import {
+  V1alpha1VirtualMachineRestore,
+  V1alpha1VirtualMachineSnapshot,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseSnapshotData = (
+  namespace: string,
+) => [V1alpha1VirtualMachineSnapshot[], Map<string, V1alpha1VirtualMachineRestore>, boolean, any];
+
+const useSnapshotData: UseSnapshotData = (namespace) => {
+  const [snapshots, snapshotsLoaded, snapshotsError] = useK8sWatchResource<
+    V1alpha1VirtualMachineSnapshot[]
+  >({
+    isList: true,
+    groupVersionKind: VirtualMachineSnapshotModelGroupVersionKind,
+    namespaced: true,
+    namespace: namespace,
+  });
+
+  const [restores] = useK8sWatchResource<V1alpha1VirtualMachineRestore[]>({
+    isList: true,
+    groupVersionKind: VirtualMachineRestoreModelGroupVersionKind,
+    namespaced: true,
+    namespace: namespace,
+  });
+
+  const restoresMap = React.useMemo(() => {
+    // we map each snapshot to its restores array
+    const map = new Map<string, V1alpha1VirtualMachineRestore>();
+    restores?.forEach((restore) => {
```

Comment:
this loop doesn't promise you will use the most recent restore.

try something like
```ts
restoresMap = restores.reduce((restoreMap, currentRestore) => {
        const relevantRestore = restoreMap[getVmRestoreSnapshotName(currentRestore)];
        if (
          !relevantRestore ||
          new Date(getVmRestoreTime(relevantRestore)).getTime() <
            new Date(getVmRestoreTime(currentRestore)).getTime()
        ) {
          restoreMap[getVmRestoreSnapshotName(currentRestore)] = currentRestore;
        }
        return restoreMap;
      }, {}),
```